src/model.rs

```diff
@@ -29,31 +29,25 @@ pub fn to_yaml<T: Serialize>(instance: &T) -> Result<String> {
 /// Will return `Err` if there is an issue converting YAML files for spec+annotation into a model
 /// instance.
 pub fn from_yaml<T: DeserializeOwned>(
-    spec_yaml: &str,
     hash: &str,
+    spec_yaml: &str,
     annotation_yaml: Option<&str>,
 ) -> Result<T> {
-    let mut spec_mapping: BTreeMap<String, Value> = serde_yaml::from_str(spec_yaml)?;
-
-    // Insert annotation if there is something
-    if let Some(yaml) = annotation_yaml {
-        let annotation_map: Mapping = serde_yaml::from_str(yaml)?;
-        spec_mapping.insert("annotation".into(), Value::from(annotation_map));
+    let mut spec: BTreeMap<String, Value> = serde_yaml::from_str(spec_yaml)?;
+    spec.insert("hash".to_owned(), Value::from(hash));
```

Comment:
We storing the hash now?

Reply:
No, this line is in `from_yaml` so it is about creating a model in memory from YAML strings.

The main reason it shows up as a diff is because I moved the pieces about `spec` to be together instead of spliting it by `annotation` and renamed the variables to simplify. `spec_mapping` on a type of `BTreeMap` was a redundant name.